assets/js/index.js

```diff
@@ -0,0 +1,255 @@
+  // fetch data from API
+  // current data url
+  const currentDataUrl = constructUrl(
+    "https://api.openweathermap.org/data/2.5/weather",
+    {
+      q: cityName,
+      appid: apiKey,
+    }
+  );
+
+  const currentData = await fetchData(currentDataUrl);
+  if (currentData) {
+    updateRecentSearch(cityName);
+  }
+
+  // get lat, lon and city name
+  const lat = currentData?.coord?.lat;
+  const lon = currentData?.coord?.lon;
+  const displayCityName = currentData?.name;
+
+  const forecastData = await getForecastData(lat, lon);
+
+  return {
+    cityName: displayCityName,
+    weatherData: forecastData,
+  };
+};
+
+const getForecastData = async (lat, lon) => {
+  // forecast url
+  const forecastDataUrl = constructUrl(
+    "https://api.openweathermap.org/data/2.5/onecall",
+    {
+      lat: lat,
+      lon: lon,
+      exclude: "minutely,hourly",
+      units: "metric",
+      appid: apiKey,
+    }
+  );
+
+  const forecastData = await fetchData(forecastDataUrl);
+
+  return forecastData;
+};
+
+// display recently searched section
+
+const displayRecentSearches = () => {
+  const recentlySearched = readFromLocalStorage("recentlySearched", []);
+  if (recentlySearched.length) {
+    $("#recentSearchList").empty();
+    recentlySearched.forEach((city) => createRecentCity(city, true));
+  } else {
+    createRecentCity(null, false);
+  }
+};
+
+const createRecentCity = (city, canDisplay) => {
+  if (canDisplay) {
+    $("#recentSearchList").append(
+      `<li type="button" data-city="${city}" class="btn btn-primary recent-btn">${city}</li>`
+    );
+    if ($("#recentPlaceholder")) {
+      $("#recentPlaceholder").remove();
+    }
+  }
+};
+
+// handle user input
+
+const handleFormSubmit = async (event) => {
+  event.preventDefault();
+  const cityName = $("#searchInput").val();
+  let reformattedCityName = "";
+
+  if (cityName) {
+    // get recent search data
+    reformattedCityName = reformatString(cityName.trim());
+  }
+
+  // call from API to get data
+  const currentForecast = await fetchWeatherData(reformattedCityName);
+
+  setCurrentData(currentForecast);
+  setForecastData(currentForecast);
+};
+
+const reformatString = (cityName) => {
+  lowerCity = cityName.toLowerCase();
+  return cityName.charAt(0).toUpperCase() + lowerCity.slice(1);
+};
+
+const updateRecentSearch = (reformattedCityName) => {
+  const recentlySearched = readFromLocalStorage("recentlySearched", []);
+
+  if (!recentlySearched.includes(reformattedCityName)) {
+    // add city to recently searched
+    recentlySearched.unshift(reformattedCityName);
+  }
+  if (recentlySearched.length > 5) {
+    recentlySearched.pop();
+  }
+
+  writeToLocalStorage("recentlySearched", recentlySearched);
+  displayRecentSearches();
+};
+
+// display data
+
+const setCurrentData = (currentData) => {
+  $("#localWeatherHeader").remove();
+  $("#currentWeatherSection").empty();
+  let now = moment().format("dddd, MMMM Do YYYY, h:mm:ss a");
```

Comment:
This should be `const` - although the time obviously changes, we're not actually changing the value, it is always equal to `moment().format("dddd, MMMM Do YYYY, h:mm:ss a");`